yb-voyager/cmd/analyzeSchema.go

```diff
@@ -374,12 +376,54 @@ func checkStmtsUsingParser(sqlInfoArr []sqlInfo, fpath string, objType string) {
 			continue
 		}
 
-		if objType == TABLE {
-			createTableNode, ok := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_CreateStmt)
-			if ok {
-				reportGeneratedStoredColumnTables(createTableNode, sqlStmtInfo, fpath)
-			}
+		createTableNode, isCreateTable := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_CreateStmt)
+		alterTableNode, isAlterTable := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_AlterTableStmt)
+		createIndexNode, isCreateIndex := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_IndexStmt)
+
+		if objType == TABLE && isCreateTable {
+			reportGeneratedStoredColumnTables(createTableNode, sqlStmtInfo, fpath)
+		}
+		if isAlterTable {
+			reportAlterTableVariants(alterTableNode, sqlStmtInfo, fpath, objType)
+		}
+		if isCreateIndex {
+			reportCreateIndexStorageParameter(createIndexNode, sqlStmtInfo, fpath)
+		}
+
+	}
+}
+
+func reportCreateIndexStorageParameter(createIndexNode *pg_query.Node_IndexStmt, sqlStmtInfo sqlInfo, fpath string) {
+	indexName := createIndexNode.IndexStmt.GetIdxname()
+	summaryMap["INDEX"].invalidCount[sqlStmtInfo.objName] = true
+	if len(createIndexNode.IndexStmt.GetOptions()) > 0 {
+		reportCase(fpath, STORAGE_PARAMETERS_DDL_STMT_ISSUE, "https://github.com/yugabyte/yugabyte-db/issues/23467",
+			"Remove the storage parameters from the DDL", "INDEX", indexName, sqlStmtInfo.stmt)
+	}
+}
+
+func reportAlterTableVariants(alterTableNode *pg_query.Node_AlterTableStmt, sqlStmtInfo sqlInfo, fpath string, objType string) {
+	schemaName := alterTableNode.AlterTableStmt.Relation.Schemaname
+	tableName := alterTableNode.AlterTableStmt.Relation.Relname
+	fullyQualifiedName := lo.Ternary(schemaName != "", schemaName+"."+tableName, tableName)
+	if objType == "TABLE" {
+		// this will the list of items in the SET (attribute=value, ..)
+		setParameters := alterTableNode.AlterTableStmt.Cmds[0].GetAlterTableCmd().GetDef().GetList()
+		if len(setParameters.GetItems()) > 0 {
+			reportCase(fpath, ALTER_TABLE_SET_ATTRUBUTE_ISSUE, "https://github.com/yugabyte/yugabyte-db/issues/1124",
+				"Remove it from the exported schema", "TABLE", fullyQualifiedName, sqlStmtInfo.stmt)
 		}
+
+		if len(alterTableNode.AlterTableStmt.Cmds[0].GetAlterTableCmd().GetDef().GetConstraint().GetOptions()) > 0 {
```

Comment:
Can you add documentation for all these type of statements here (sample statement, pg_parser documentation of how to get the details we want, etc), so that we clearly know which types we are searching for and reporting..

Reply:
Added example stmt, and description of how and why we are checking this info. I can't find any proper documentation for this. Let me know if I should add more information in these comments